components/CommentPopup.js

```diff
@@ -0,0 +1,151 @@
+import { useState } from "react";
+import styled from "styled-components";
+
+const CommentPopup = ({ show, onClose, onSubmit, commentToEdit, onDelete }) => {
+  const [comment, setComment] = useState(
+    commentToEdit ? commentToEdit.text : ""
+  );
+
+  const handleSubmit = () => {
+    if (comment.trim()) {
+      onSubmit(comment, commentToEdit ? commentToEdit.id : null);
+      setComment("");
+      onClose();
+    }
+  };
+
+  if (!show) return null;
+
+  return (
+    <Overlay>
+      <Popup>
+        <Header>
+          <Title>{commentToEdit ? "Edit Comment" : "Comment"}</Title>
+          <CloseButton onClick={onClose}>×</CloseButton>
+        </Header>
+        <Content>
+          <TextArea
+            value={comment}
+            onChange={(event) => setComment(event.target.value)}
+            placeholder="Write your comment..."
+          />
+        </Content>
+        <Footer>
+          {commentToEdit && (
+            <DeleteButton onClick={() => onDelete(commentToEdit.id)}>
+              Delete
+            </DeleteButton>
+          )}
+          <SubmitButton onClick={handleSubmit} disabled={!comment.trim()}>
+            {commentToEdit ? "Update" : "Submit"}
+          </SubmitButton>
+        </Footer>
+      </Popup>
+    </Overlay>
+  );
+};
+
+export default CommentPopup;
+
+const Overlay = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background: rgba(0, 0, 0, 0.7);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const Popup = styled.div`
```

Comment:
this could be a section element and thus be more semantic.

Reply:
@NikRoe  is this referred to the Overlay or to the Popup?

Reply:
Gotcha! :D